### PR TITLE
feat(loader): Sync loader with JS monorepo

### PR DESCRIPTION
**Pull request by AbhiPrasad**
As per https://github.com/getsentry/sentry-javascript/issues/6990 the loader in the JS monorepo and sentry template has drifted apart. This syncs them together (validated by the integration tests in the JS monorepo).

PR in JS Repo: https://github.com/getsentry/sentry-javascript/pull/7001

### CODE REVIEW

File: src/sentry/templates/sentry/js-sdk-loader.js.tmpl [hunk 1]
```diff
@@ -29,8 +29,8 @@
     // content.p = promise rejection
     // content.f = function call the Sentry
     if (
-      (content.e ||
-        content.p ||
+      ('e' in content ||
+        'p' in content ||
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/js-sdk-loader.js.tmpl:33]: This was from https://github.com/getsentry/sentry-javascript/pull/2207

File: src/sentry/templates/sentry/js-sdk-loader.js.tmpl [hunk 2]
```diff
@@ -56,7 +56,7 @@
     // come out in the wrong order. Because of that we don't need async=1 as GA does.
     // it was probably(?) a legacy behavior that they left to not modify few years old snippet
     // https://www.html5rocks.com/en/tutorials/speed/script-loading/
-    var _currentScriptTag = _document.getElementsByTagName(_script)[0];
+    var _currentScriptTag = _document.scripts[0];
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/js-sdk-loader.js.tmpl:59]: https://github.com/getsentry/sentry-javascript/pull/3578

File: src/sentry/templates/sentry/js-sdk-loader.js.tmpl [hunk 3]
```diff
@@ -68,6 +68,9 @@
         _window[_onerror] = _oldOnerror;
         _window[_onunhandledrejection] = _oldOnunhandledrejection;
 
+        // Add loader as SDK source
+        _window.SENTRY_SDK_SOURCE = 'loader';
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/js-sdk-loader.js.tmpl:72]: https://github.com/getsentry/sentry-javascript/pull/6985

File: src/sentry/templates/sentry/js-sdk-loader.js.tmpl [hunk 4]
```diff
@@ -137,9 +140,9 @@
 
       // And now capture all previously caught exceptions
       for (var i = 0; i < data.length; i++) {
-        if (data[i].e && tracekitErrorHandler) {
+        if ('e' in data[i] && tracekitErrorHandler) {
           tracekitErrorHandler.apply(_window, data[i].e);
-        } else if (data[i].p && tracekitUnhandledRejectionHandler) {
+        } else if ('p' in data[i] && tracekitUnhandledRejectionHandler) {
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/js-sdk-loader.js.tmpl:145]: https://github.com/getsentry/sentry-javascript/pull/2207

File: src/sentry/templates/sentry/js-sdk-loader.js.tmpl [hunk 5]
```diff
@@ -198,9 +201,9 @@
 
   // Do the same store/queue/call operations for `onunhandledrejection` event
   var _oldOnunhandledrejection = _window[_onunhandledrejection];
-  _window[_onunhandledrejection] = function(exception) {
+  _window[_onunhandledrejection] = function(e) {
     queue({
-      p: exception.reason
+      p: 'reason' in e ? e.reason : 'detail' in e && 'reason' in e.detail ? e.detail.reason : e
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/js-sdk-loader.js.tmpl:206]: https://github.com/getsentry/sentry-javascript/issues/2429

**Review by kamilogorek — APPROVED** [review]

**Comment by kamilogorek** [comment]
LGTM!

**Pull request merged**